java/spark/src/main/java/com/logicalclocks/hsfs/spark/engine/StatisticsEngine.java

```diff
@@ -52,65 +55,69 @@ public StatisticsEngine(EntityEndpointType entityType) {
-    Optional<Statistics> statistics = computeStatistics(dataFrame,
+    Statistics statistics = computeStatistics(dataFrame,
         trainingDataset.getStatisticsConfig().getColumns(),
         trainingDataset.getStatisticsConfig().getHistograms(),
         trainingDataset.getStatisticsConfig().getCorrelations(),
         trainingDataset.getStatisticsConfig().getExactUniqueness(),
         null);
 
-    if (statistics.isPresent()) {
-      return statisticsApi.post(trainingDataset, statistics.get());
-    }
-
-    return null;
+    return statisticsApi.post(trainingDataset, statistics);
   }
 
   public Statistics computeStatistics(FeatureView featureView, TrainingDataset trainingDataset, Dataset<Row> dataFrame)
       throws FeatureStoreException, IOException {
-    Optional<Statistics> statistics = computeStatistics(dataFrame,
+    Statistics statistics = computeStatistics(dataFrame,
         trainingDataset.getStatisticsConfig().getColumns(),
         trainingDataset.getStatisticsConfig().getHistograms(),
         trainingDataset.getStatisticsConfig().getCorrelations(),
         trainingDataset.getStatisticsConfig().getExactUniqueness(),
         null);
 
-    if (statistics.isPresent()) {
-      return statisticsApi.post(featureView, trainingDataset.getVersion(), statistics.get());
-    }
-
-    return null;
+    return statisticsApi.post(featureView, trainingDataset.getVersion(), statistics);
   }
 
   public Statistics computeStatistics(FeatureGroupBase featureGroup, Dataset<Row> dataFrame, Long commitId)
       throws FeatureStoreException, IOException {
 
-    Optional<Statistics> statistics = computeStatistics(dataFrame,
+    Statistics statistics = computeStatistics(dataFrame,
         featureGroup.getStatisticsConfig().getColumns(),
         featureGroup.getStatisticsConfig().getHistograms(),
         featureGroup.getStatisticsConfig().getCorrelations(),
         featureGroup.getStatisticsConfig().getExactUniqueness(),
         commitId);
 
-    if (statistics.isPresent()) {
-      return statisticsApi.post(featureGroup, statistics.get());
-    }
-
-    return null;
+    return statisticsApi.post(featureGroup, statistics);
   }
 
-  private Optional<Statistics> computeStatistics(Dataset<Row> dataFrame, List<String> statisticColumns,
+  private Statistics computeStatistics(Dataset<Row> dataFrame, List<String> statisticColumns,
                                                  Boolean histograms,
                                                  Boolean correlations, Boolean exactUniqueness, Long commitId) {
+    String content;
+    Long commitTime = Timestamp.valueOf(LocalDateTime.now()).getTime();
+
     if (dataFrame.isEmpty()) {
       LOGGER.warn("There is no data in the entity that you are trying to compute statistics for. A "
           + "possible cause might be that you inserted only data to the online storage of a feature group.");
-      return Optional.empty();
+
+      JSONArray columns = new JSONArray();
```

Comment:
Can you refactor it as its own function ?

Reply:
Done.